tests/Unit/Routing/PagedRouteCollectionTest.php

```diff
@@ -47,12 +47,18 @@ public function testIterator($amountRoutes, $routesLoadedInParallel, $expectedCa
         }
         $names = array_keys($routes);
 
+        $counter = 0;
+        $with = [];
+        $will = [];
         foreach ($expectedCalls as $i => $range) {
-            $this->routeProvider->expects($this->at($i))
-              ->method('getRoutesPaged')
-              ->with($range[0], $range[1])
-              ->will($this->returnValue(array_slice($routes, $range[0], $range[1])));
+            $with[] = [$range[0], $range[1]];
+            $will[] = array_slice($routes, $range[0], $range[1]);
+            ++$counter;
         }
+        $mocker = $this->routeProvider->expects($this->exactly($counter))
+            ->method('getRoutesPaged');
+        \call_user_func_array([$mocker, 'withConsecutive'], $with);
```

Comment:
if i am not mistaken, we can use `$this->routeProvider` here. `$mocker` is just a reference to the same object, as the phpunit methods return the object itself to allow chaining. can you please remove the extra variable? makes the code easier to read imho.

Reply:
Checked and reverted because `\PHPUnit\Framework\MockObject\Builder\InvocationMocker::method()` returns is different mocker